### PR TITLE
use standard error when the entity id cannot be used

### PR DESCRIPTION
**Pull request by mhilton**

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1276/
Test PASSed.

File: internal/charmstore/store.go [hunk 1]
```diff
@@ -513,8 +513,11 @@ type AddParams struct {
...
 }
 
-// AddCharm adds a charm entities collection with the given
-// parameters.
+// AddCharm adds a charm entities collection with the given parameters.
+// If p.URL cannot be used as a name for the charm then the returned
+// error will have the cause params.ErrEntityIdNotAllowed. If the charm
+// duplicates an existing charm then the returned error will have the
+// cause params.ErrDuplicateUpload.
 func (s *Store) AddCharm(c charm.Charm, p AddParams) (err error) {
```
Comment (urosj) [inline internal/charmstore/store.go:521]: Hm, we don't actually return this ErrDuplicateUpload error eny more, do we?

File: internal/charmstore/store.go [hunk 2]
```diff
@@ -1011,7 +1014,10 @@ func baseURL(url *charm.URL) *charm.URL {
 var errNotImplemented = errgo.Newf("not implemented")
 
 // AddBundle adds a bundle to the entities collection with the given
-// parameters.
+// parameters. If p.URL cannot be used as a name for the bundle then the
+// returned error will have the cause params.ErrEntityIdNotAllowed. If
+// the bundle duplicates an existing bundle then the returned error will
+// have the cause params.ErrDuplicateUpload.
```
Comment (urosj) [inline internal/charmstore/store.go:1020]: Same as above?

File: internal/charmstore/store.go (hunk 1 above)
Comment (mhilton) [inline internal/charmstore/store.go:521]: Yes it will come from insertEntity

**Comment by urosj** [comment]
:+1:

**Comment by frankban** [comment]
:+1: thank you!

**Comment by mhilton** [comment]
:shipit: thanks for the reviews

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Pull request merged**